### PR TITLE
Fixes #15502: Add autocompletion to rudder-pkg

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/15502

### CODE REVIEW

File: relay/sources/Makefile [hunk 1]
```diff
@@ -47,19 +47,26 @@ relay-api/virtualenv/virtualenv.py: relay-api/virtualenv.tgz
...
-man-source: target/man-source/rudder-relayd.1;
+man-source: target/man-source/rudder-relayd.1 autocomplete-source;
 
 target/man-source/%: man/%.adoc
 	mkdir -p target/man-source
 	asciidoctor -D target/man-source -b manpage $<
 
-man: target/man/rudder-relayd.1.gz;
+autocomplete-source:
+	PYTHONPATH=$(CURDIR)/rudder-pkg/lib/rudder-pkg:$$PYTHONPATH; cd man && docopt-completion $(CURDIR)/rudder-pkg/rudder-pkg --manual-bash && cd -
```
Comment (amousset) [inline relay/sources/Makefile]: would it be possible to generate this file in the target directory instead? (and created file path as target)

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

File: relay/sources/Makefile (hunk 1 above)
Comment (Fdall) [reply relay/sources/Makefile]: fixed

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

File: relay/sources/Makefile [hunk 2]
```diff
@@ -47,19 +47,27 @@ relay-api/virtualenv/virtualenv.py: relay-api/virtualenv.tgz
 rudder-pkg/rudder_plugins_key.pub:
 	$(GET) rudder-pkg/rudder_plugins_key.pub https://repository.rudder.io/apt/rudder_apt_key.pub $(RUDDER_PLUGINS_KEY_SHA256)
 
-man-source: target/man-source/rudder-relayd.1;
+man-source: target/man-source/rudder-relayd.1 autocomplete-source/man/rudder-pkg.sh;
```
Comment (peckpeck) [inline relay/sources/Makefile]: This doesn't match

File: relay/sources/Makefile [hunk 3]
```diff
@@ -47,19 +47,27 @@ relay-api/virtualenv/virtualenv.py: relay-api/virtualenv.tgz
...
 
 target/man-source/%: man/%.adoc
 	mkdir -p target/man-source
 	asciidoctor -D target/man-source -b manpage $<
 
-man: target/man/rudder-relayd.1.gz;
+target/autocomplete-source/man/rudder-pkg.sh:
```
Comment (peckpeck) [inline relay/sources/Makefile]: that

File: relay/sources/Makefile [hunk 4]
```diff
@@ -47,19 +47,27 @@ relay-api/virtualenv/virtualenv.py: relay-api/virtualenv.tgz
...
+	PYTHONPATH=$(CURDIR)/rudder-pkg/lib/rudder-pkg:$$PYTHONPATH; cd man && docopt-completion $(CURDIR)/rudder-pkg/rudder-pkg --manual-bash && cd -
+
+man: target/man/rudder-relayd.1.gz target/man/rudder-pkg.sh;
 
 target/man/rudder-relayd.1.gz:
 	mkdir -p target/man/
 	$(GET) target/man/rudder-relayd.1 https://docs.rudder.io/man/${RUDDER_MAJOR_VERSION}/rudder-relayd.1 ""
 	gzip -f target/man/rudder-relayd.1
 
+target/man/rudder-pkg.sh:
```
Comment (peckpeck) [inline relay/sources/Makefile]: there should not be a manual for this thing

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
OK, merging this PR

**Pull request merged**